integration/networking/resolvconf_test.go

```diff
@@ -0,0 +1,72 @@
+package networking
+
+import (
+	"os"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/network"
+
+	"github.com/docker/docker/testutil/daemon"
+	"gotest.tools/v3/skip"
+)
+
+// Regression test for https://github.com/moby/moby/issues/46968
+func TestResolvConfLocalhostIPv6(t *testing.T) {
+	// No "/etc/resolv.conf" on Windows.
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+	// In rootless mode, "/etc/resolv.conf" isn't the host's, so it can't be replaced by the test.
+	skip.If(t, testEnv.IsRootless())
+
+	ctx := setupTest(t)
+
+	// Replace the test host's resolv.conf with one that only contains a loopback address.
+	// TODO(robmry) - there must be a better way!
+	realRC, err := os.ReadFile("/etc/resolv.conf")
+	assert.NilError(t, err)
+	defer func() {
+		os.WriteFile("/etc/resolv.conf", realRC, 0644)
+	}()
+	os.WriteFile("/etc/resolv.conf", []byte("nameserver 127.0.0.53"), 0644)
+
+	d := daemon.New(t)
+	d.StartWithBusybox(ctx, t, "--experimental", "--ip6tables")
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	netName := "nnn"
+	network.CreateNoError(ctx, t, c, netName,
+		network.WithDriver("bridge"),
+		network.WithIPv6(),
+		network.WithIPAM("fd49:b5ef:36d9::/64", "fd49:b5ef:36d9::1"),
+	)
+	defer network.RemoveNoError(ctx, t, c, netName)
+
+	result := container.RunAttach(ctx, t, c,
+		container.WithImage("busybox:latest"),
+		container.WithNetworkMode(netName),
+		container.WithCmd("cat", "/etc/resolv.conf"),
+	)
+	defer c.ContainerRemove(ctx, result.ContainerID, containertypes.RemoveOptions{
+		Force: true,
+	})
+
+	assert.Check(t, is.Equal(result.Stdout.String(), `# Generated by Docker Engine.
```

Comment:
This might be a good place to use https://pkg.go.dev/gotest.tools/v3@v3.5.1/golden instead of having to update the expected string manually whenever the template is changed.

Reply:
For this single integration test, I think it's probably more fiddly to work out how to run the test with "-update" to generate the expected output than to just replace the text.

But I have now used it for the 40-ish unit tests in `libnetwork/internal/resolvconf`.